client/cypress/integration/dashboard/parameter_mapping_spec.js

```diff
@@ -24,32 +29,67 @@ describe("Parameter Mapping", () => {
       });
   });
 
-  const openMappingOptions = (widgetTestId, paramName) => {
+  const openMappingOptions = widgetTestId => {
     cy.getByTestId(widgetTestId).within(() => {
       cy.getByTestId("WidgetDropdownButton").click();
     });
 
     cy.getByTestId("WidgetDropdownButtonMenu")
       .contains("Edit Parameters")
       .click();
-
-    cy.getByTestId(`EditParamMappingButton-${paramName}`).click();
   };
 
-  const saveMappingOptions = () => {
-    cy.getByTestId("EditParamMappingPopover").within(() => {
+  const saveMappingOptions = (close = true) => {
+    cy.getByTestId("EditParamMappingPopover")
+      .filter(":visible")
+      .within(() => {
+        cy.contains("button", "OK").click();
+      });
+
+    if (close) {
       cy.contains("button", "OK").click();
-    });
+    }
+  };
+
+  const reorderTwoParameters = (param1, param2, type = "WIDGET") => {
+    cy.get(".parameter-block")
+      .first()
+      .invoke("width")
+      .as("paramWidth");
+
+    const dragParam = (paramName, offsetLeft, offsetTop) => {
```

Comment:
Comments for Cypress:
1. I think now it makes sense to rename this file to `parameter_spec.js` (or create a new one at will) and separate the `describe("Parameter Mapping")` from the `describe("Parameter Dragging")`. Dragging should have enough scope to have its own test 😬
2. This method is the same as the one from queries, right? It's better if we create a `support/parameters.js` to put related helper functions and avoid the duplicate (don't worry if it doesn't have a folder yet, I'll soon refactor those helper functions' location)

Reply:
The method is not quite the same, but similar. I'll try to adapt them to something more general